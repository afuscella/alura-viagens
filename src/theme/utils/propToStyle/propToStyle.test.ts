import { propToStyle } from './index';

const propToStyleTextAlign = propToStyle('textAlign');
const propToStyleFlex = propToStyle('flex');

describe('propToStyle()', () => {
  describe('when propToStyle receives an sample argument', () => {
    test('and its is a string, should convert to style', () => {
      // <Text textAlign="center" />
      const componentPropsExp = { textAlign: 'center' }; // string
      const componentPropsAct = propToStyleTextAlign(componentPropsExp);

      expect(componentPropsExp).toEqual(componentPropsAct);
    });

    test('and its is a number, should convert to style', () => {
      // <Box flex={1} />
      const componentPropsExp = { flex: 1 }; // number
      const componentPropsAct = propToStyleFlex(componentPropsExp);

      expect(componentPropsExp).toEqual(componentPropsAct);
    });
  });

  describe('when propToStyle receives an argument with breakpoint', () => {
    test('render only a single small breakpoint resolution', () => {
      // <Text textAlign={{ xs: 'center' }} />
      const componentProps = { textAlign: { xs: 'center' } };
      const componentPropsAct = propToStyleTextAlign(componentProps);

      expect(componentPropsAct).toMatchSnapshot();
    });

    test('render only a single extra large breakpoint resolution', () => {
      // <Text textAlign={{ xl: 'center' }} />
      const componentProps = { textAlign: { xl: 'center' } };
      const componentPropsAct = propToStyleTextAlign(componentProps);

      expect(componentPropsAct).toMatchSnapshot();
    });
  });

  describe('when propToStyle receives an argument with breakpoints', () => {
    test('render multiple breakpoints resolution', () => {
      // <Text textAlign={{ sm: 'left', md: 'center', lg: 'right' }} />
      const componentProps = { textAlign: { sm: 'left', md: 'center', lg: 'right' } };
      const componentPropsAct = propToStyleTextAlign(componentProps);

      expect(componentPropsAct).toMatchSnapshot();
    });
  });
});
