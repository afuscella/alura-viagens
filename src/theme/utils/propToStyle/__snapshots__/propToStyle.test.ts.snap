// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when propToStyle receives an argument with breakpoint render only a single extra large breakpoint resolution 1`] = `Array []`;

exports[`propToStyle() when propToStyle receives an argument with breakpoint render only a single small breakpoint resolution 1`] = `
Array [
  "@media only screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  "}",
  "@media only screen and (min-width:",
  "400",
  "px){",
  "}",
  "@media only screen and (min-width:",
  "768",
  "px){",
  "}",
  "@media only screen and (min-width:",
  "992",
  "px){",
  "}",
  "@media only screen and (min-width:",
  "1120",
  "px){",
  "}",
]
`;

exports[`propToStyle() when propToStyle receives an argument with breakpoints render multiple breakpoints resolution 1`] = `Array []`;
